docs/guides/developers.md

```diff
@@ -0,0 +1,45 @@
+# Developer guide
+This guide contains some more in-depth guidelines, tasks and examples that developers should adhere to when working on hydrolib-core.
+In addition to being a useful resource for developing, this documentation can also be helpful for code reviews.
+
+## Explicitly exposing the public API
+Whenever you add new functionality that should (or could) be used by the users, make sure you explicitly expose them in the correct `__init__.py` files.
+
+For example, let's say you have implemented a new `TimModel` that represents a `.tim` file. You have also added a new `TimParser` and `TimSerializer` to read and write from and to `.tim` files. The `TimModel` is a class that should be publicly exposed to the user so that they can use it to represent `.tim` files. The parser and serializer, however, do not need to be exposed, as these are implementation details that we do not want to bother our users with. To expose the `TimModel` explicitly for our users we have to update both the `hydrolib/core/dflowfm/tim/__init__.py` (assuming you created a new `tim` folder for this new functionality) and the `hydrolib/core/dflowfm/__init__.py`:
+
+```python
+# hydrolib/core/dflowfm/tim/__init__.py
+from .models import TimModel
+
+__all__ = [
+    "TimModel",
+]
+```
+
+```python
+# hydrolib/core/dflowfm/__init__.py
+...
+...
+from .tim import *
+```
+
+## Updating the documentation
+In the hydrolib-core repository there are several reference files that are used to automatically generate the [API reference](../reference/api.md). If you add or make changes to the existing API, always make sure you check if the API reference is still up to date.
```

Comment:
If you add or make changes to the existing API, always make sure you check if the API reference is still up to date.

Can be changed into:

If you add or make changes to the existing API, always make sure the API reference is still up to date.

Reply:
Done